### PR TITLE
Chap01 Ex2a

### PR DESCRIPTION
**Pull request by thefundamentaltheor3m**
Long proof that can be shortened… any suggestions would be appreciated

### CODE REVIEW

File: src/chapter01/exercises/exercise02.lean [hunk 1]
```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
...
-(h2 : X ⊆ B) : sorry :=
+(h2 : X ⊆ B) : X ≠ D :=
 begin
-  sorry
-end
+  intro hX',
+  rw hX' at *,
+  have hD : ¬ (E ⊆ D),
+  { have h0 : D = ({0} : set ℝ),
+    -- Any way I can get this hypothesis from the previous proof?
```
Comment (kbuzzard) [inline src/chapter01/exercises/exercise02.lean]: No you can't get it from the previous proof. Proofs are erased after being checked. So what you should do is to make a new lemma saying D = {0} and prove it before part_a. Then you'll be able to use the lemma in both part a and part b.

File: src/chapter01/exercises/exercise02.lean [hunk 2]
```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
 -- Which pair of these sets has the property that neither is contained in the other?
 lemma parta : ∃ S T ∈ ({B, C, D, E} : set (set ℝ)), ¬ (S ⊆ T) ∧ ¬ (T ⊆ S) :=
 begin
-  sorry
+  have h0 : D = ({0} : set ℝ),
+  { simp_rw D,
+    ext,
+    split,
+    { intro hx,
+      simp at hx,
```
Comment (kbuzzard) [inline src/chapter01/exercises/exercise02.lean]: You don't need to do this `simp` call, it doesn't actually do much. Use `squeeze_simp` to see what it's doing, but it's just changing x^2<1 to |x|<1 and then rearranging stuff. I would put this off until later.

File: src/chapter01/exercises/exercise02.lean [hunk 3]
```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
...
+      simp,
```
Comment (kbuzzard) [inline src/chapter01/exercises/exercise02.lean]: Don't do non-terminal `simp`s. Use `squeeze_simp` to find out the name of the lemma or lemmas being used, and then rewrite them explicitly.

File: src/chapter01/exercises/exercise02.lean [hunk 4]
```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
...
+      cases hx with hl hr,
+      cases hl with a ha,
```
Comment (kbuzzard) [inline src/chapter01/exercises/exercise02.lean]: Don't do cases, cases; use `rcases` to take the term apart all in one go.

File: src/chapter01/exercises/exercise02.lean [hunk 5]
```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
...
+      rw ha at hr,
+      rw ha,
```
Comment (kbuzzard) [inline src/chapter01/exercises/exercise02.lean]: Don't be rewriting `ha` everywhere. Use the `rcases rfl` trick to *define* `x` to be `\u a`.

**Review by kbuzzard — CHANGES_REQUESTED** [review]
Thanks a lot! My main comment is to factor out the lemma that D={0} rather than proving it twice.

**Comment by thefundamentaltheor3m** [comment]
> Thanks a lot! My main comment is to factor out the lemma that D={0} rather than proving it twice.

Thank you for the suggestions! I'll start working on them right away

**Comment by kbuzzard** [comment]
Can you remove the three other files in the commit (the binary .DS_store file and the two files from other chapters?)

**Comment by kbuzzard** [comment]
PS another UG sniped your part (a)!

**Comment by kbuzzard** [comment]
Someone else now did this problem. Sorry! :-(

**Pull request closed without merging**